cvxpy/lin_ops/canon_backend.py

```diff
@@ -1031,3 +1315,110 @@ def add_dicts(a: dict, b: dict) -> dict:
+        """
+        assert self.tensor is not None
+        tensor_representations = []
+        for variable_id, variable_tensor in self.tensor.items():
+            for parameter_id, parameter_tensor in variable_tensor.items():
+                for param_slice_offset in range(parameter_tensor.shape[0]):
+                    matrix = parameter_tensor[param_slice_offset]
+                    tensor_representations.append(TensorRepresentation(
+                        matrix.flatten(order='F'),
+                        np.tile(np.arange(matrix.shape[0]), matrix.shape[1]) + row_offset,
+                        np.repeat(np.arange(matrix.shape[1]), matrix.shape[0]) + self.id_to_col[variable_id],
+                        np.ones(matrix.size) * self.param_to_col[parameter_id] +
+                        param_slice_offset,
+                    ))
+        return TensorRepresentation.combine(tensor_representations)
+
+    def select_rows(self, rows: np.ndarray) -> None:
+
+        def func(x):
+            return x[:, rows, :]
+
+        self.apply_all(func)
+
+    def apply_all(self, func: Callable) -> None:
+        self.tensor = {var_id: {k: func(v)
+                                for k, v in parameter_repr.items()}
+                       for var_id, parameter_repr in self.tensor.items()}
+
+    def create_new_tensor_view(self, variable_ids: set[int], tensor: Any,
+                               is_parameter_free: bool) -> NumpyTensorView:
+        return NumpyTensorView(variable_ids, tensor, is_parameter_free, self.param_size_plus_one,
+                               self.id_to_col, self.param_to_size, self.param_to_col,
+                               self.var_length)
+
+    def accumulate_over_variables(self, func: Callable, is_param_free_function: bool) \
+            -> NumpyTensorView:
+        """
+        Apply 'func' to A and b.
+        If 'func' is a parameter free function, then we can apply it to all parameter slices
+        (including the slice that contains non-parameter constants).
+        If 'func' is not a parameter free function, we only need to consider the parameter slice
+        that contains the non-parameter constants, due to DPP rules.
+        """
+        for variable_id, tensor in self.tensor.items():
+            self.tensor[variable_id] = self.apply_to_parameters(func, tensor) if \
+                is_param_free_function else func(tensor[Constant.ID.value])
+
+        self.is_parameter_free = self.is_parameter_free and is_param_free_function
+        return self
+
+    @staticmethod
+    def apply_to_parameters(func: Callable,
+                            parameter_representation: dict[int, np.ndarray]) \
+            -> dict[int, np.ndarray]:
+        """
+        Apply 'func' to each slice of the parameter representation.
+        """
+        return {k: func(v) for k, v in parameter_representation.items()}
+
+    @staticmethod
+    def combine_potentially_none(a: Any | None, b: Any | None) -> Any | None:
```

Comment:
@Transurgeon We should add a new abstract class called `DictTensorView` that includes the functions that are (almost) identical between the two current implementation. 
The SciPy and Numpy Views would only need to implement what it means to add two tensors.
Thus, `DictTensorView` has the abstractmethod `add_tensors(a,b)`, which for SciPy is
`[a + b for a, b in zip(a, b)]`
and simply
`a + b` for Numpy.